### PR TITLE
Simplify conditional code around instrumentation feature

### PR DESCRIPTION
**Pull request by thomaseizinger**
By splitting things into modules with identical APIs, we can remove a lot of `cfg(feature)` attributes.

### CODE REVIEW

**Review by Restioson — APPROVED** [review]
Thanks!

**Comment by thomaseizinger** [comment]
I'll merge this once the dependent PR is merged.

**Pull request merged**
